src/engine/controls/bpmcontrol.cpp

```diff
@@ -936,34 +937,35 @@ double BpmControl::getBeatMatchPosition(
 
         // TODO(rryan): Revise this with something that keeps a broader number of
         // cases in sync. This at least prevents breaking out of the loop.
-        if (loop_enabled && dThisPosition <= loop_end_position) {
-            const double loop_length = loop_end_position - loop_start_position;
-            const double end_delta = dNewPlaypos - loop_end_position;
+        if (loop_enabled && thisPosition <= loopEndPositionFrames) {
+            const mixxx::FrameDiff_t loopLength = loopEndPositionFrames - loopStartPositionFrames;
+            const mixxx::FrameDiff_t endDelta = newPlaypos - loopEndPositionFrames;
 
             // Syncing to after the loop end.
-            if (end_delta > 0 && loop_length > 0.0) {
-                int i = end_delta / loop_length;
-                dNewPlaypos = loop_start_position + end_delta - i * loop_length;
+            if (endDelta > 0 && loopLength > 0.0) {
+                int i = endDelta / loopLength;
+                newPlaypos = loopStartPositionFrames + endDelta - i * loopLength;
 
                 // Move new position after loop jump into phase as well.
                 // This is a recursive call, called only twice because of
                 // respectLoops = false
-                dNewPlaypos =
-                        getNearestPositionInPhase(dNewPlaypos, false, playing);
+                newPlaypos =
+                        getNearestPositionInPhase(newPlaypos, false, playing);
             }
 
             // Note: Syncing to before the loop beginning is allowed, because
             // loops are catching
         }
     }
-    return dNewPlaypos;
+    return newPlaypos;
 }
 
 mixxx::FrameDiff_t BpmControl::getPhaseOffset(mixxx::FramePos thisPosition) {
     // This does not respect looping
-    double dNewPlayposSamples = getNearestPositionInPhase(
-            thisPosition.getValue() * kSamplesPerFrame, false, false);
-    return dNewPlayposSamples / kSamplesPerFrame - thisPosition.getValue();
+    mixxx::FramePos newPlayposFrames = getNearestPositionInPhase(
+            thisPosition, false, false);
+    qWarning() << "new play pos samples" << newPlayposFrames;
```

Comment:
It looks like you added this. Did you intend to commit it?

Reply:
My bad. Removed it in a later commit.